xesmf/smm.py

```diff
@@ -31,14 +32,22 @@ def read_weights(filename, n_in, n_out):
     A : scipy sparse COO matrix.
 
     '''
-    ds_w = xr.open_dataset(filename)
-
-    col = ds_w['col'].values - 1  # Python starts with 0
-    row = ds_w['row'].values - 1
-    S = ds_w['S'].values
-
-    weights = sps.coo_matrix((S, (row, col)), shape=[n_out, n_in])
-    return weights
+    if isinstance(weights, (str, Path, xr.Dataset)):
```

Comment:
maybe needs "idiot-proofing" like checking Path point to a netcdf file, keys are present in dict,...

Reply:
Done.